feature/widget/unread/src/main/kotlin/app/k9mail/feature/widget/unread/BaseUnreadWidgetProvider.kt

```diff
@@ -68,6 +71,7 @@ abstract class BaseUnreadWidgetProvider : AppWidgetProvider(), EarlyInit {
         for (appWidgetId in appWidgetIds) {
             repository.deleteWidgetConfiguration(appWidgetId)
         }
+        widgetScope.cancel()
```

Comment:
Canceling `widgetScope` here is unnecessary.

- Widgets actions use broadcasts that are serialized, i.e. `onDelete()` will not be called while `onUpdate()` is active.
- A new instance of this class is created for every broadcast that is received. So there will never be a situation where `onDelete()` is being called on the same instance that `onUpdate()` has been called on earlier.

Side note: `onDelete()` gets a list of app widget IDs that are being deleted. That's not necessarily all of them. So if all of the above was different, unconditionally canceling the coroutine scope here could lose updates to widgets that haven't been deleted.